### PR TITLE
deletion of registeredcluster

### PR DESCRIPTION
**Pull request by leena-jawale**
Signed-off-by: Leena Jawale <ljawale@redhat.com>

/hold

### CODE REVIEW

File: controllers/cluster-registration/cluster_registration_controller.go [hunk 1]
```diff
@@ -179,6 +197,9 @@ func (r *RegisteredClusterReconciler) getManagedCluster(regCluster *singaporev1a
 		return managedClusterList.Items[0], nil
 	}
 
+	if regCluster.DeletionTimestamp != nil {
```
Comment (robinbobbitt) [inline controllers/cluster-registration/cluster_registration_controller.go:200]: Not sure about this check? It seems misleading that a deleted registered cluster would end up returning a "correct managed cluster not found" error message. Could you share some more detail on what this is for? Thanks!

File: controllers/cluster-registration/cluster_registration_controller.go [hunk 2]
```diff
@@ -309,13 +332,102 @@ func (r *RegisteredClusterReconciler) syncManagedServiceAccount(regCluster *sing
 	return nil
 }
 
+func (r *RegisteredClusterReconciler) processRegclusterDeletion(regCluster *singaporev1alpha1.RegisteredCluster, managedCluster *clusterapiv1.ManagedCluster, hubCluster *helpers.HubInstance) (ctrl.Result, error) {
```
Comment (robinbobbitt) [inline controllers/cluster-registration/cluster_registration_controller.go:335]: Do we need to also delete the import command configmap and kubeconfig secret or are those automatically cleaned up? The kubeconfig secret has the owner set but it doesn't look like the configmap does.

**Review by robinbobbitt — COMMENTED** [review]
Looks great @leena-jawale ! Added a few questions/comments. Thanks!

File: controllers/cluster-registration/cluster_registration_controller.go (hunk 2 above)
Comment (leena-jawale) [reply controllers/cluster-registration/cluster_registration_controller.go:335]: I am adding ownerref to configmap in my ClusterID status PR.

File: controllers/cluster-registration/cluster_registration_controller.go (hunk 1 above)
Comment (leena-jawale) [reply controllers/cluster-registration/cluster_registration_controller.go:200]: Added this check, if regcluster is deleted and in deletion process managedcluster is deleted. In this case it was returning error message correct managed cluster is not found

Comment (robinbobbitt) [reply controllers/cluster-registration/cluster_registration_controller.go:200]: Ah I had this backwards in my head. Thanks for clarifying!

**Pull request merged**